src/seedu/addressbook/storage/AddressBookDecoder.java

```diff
@@ -0,0 +1,90 @@
+package seedu.addressbook.storage;
+
+import static seedu.addressbook.parser.Parser.PERSON_DATA_ARGS_FORMAT;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.Address;
+import seedu.addressbook.data.person.Email;
+import seedu.addressbook.data.person.Name;
+import seedu.addressbook.data.person.Person;
+import seedu.addressbook.data.person.Phone;
+import seedu.addressbook.data.person.UniquePersonList;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.storage.StorageFile.StorageOperationException;
+
+/**
+ * Decodes the storage data file into an {@code AddressBook} object.
+ */
+public class AddressBookDecoder {
+
+    /**
+     * Decodes {@code encodedAddressBook} into an {@code AddressBook} containing the decoded persons.
+     *
+     * @throws IllegalValueException if any of the fields in any encoded person string is invalid.
+     * @throws StorageOperationException if the {@code encodedAddressBook} is not in an invalid format.
```

Comment:
:o i think you meant "if {encodedAddressBook} is ~~not~~ in an invalid format."

Reply:
oh crap... 😓